server/storage/mvcc/kvstore.go

```diff
@@ -255,7 +232,8 @@ func (s *store) compact(trace *traceutil.Trace, rev int64) (<-chan struct{}, err
 		start := time.Now()
 		keep := s.kvindex.Compact(rev)
 		indexCompactionPauseMs.Observe(float64(time.Since(start) / time.Millisecond))
-		if !s.scheduleCompaction(rev, keep) {
+		_, err := s.scheduleCompaction(rev, keep)
+		if err != nil {
```

Comment:
```suggestion
		if _, err := s.scheduleCompaction(rev, keep); err != nil {
```

Reply:
Previously it returned bool, whether compaction succeeded (or was interrupted by stopC). I have changed it to error as it better represents intention.